src/main/scala/mesosphere/marathon/event/HistoryActor.scala

```diff
@@ -20,6 +23,8 @@ class HistoryActor(eventBus: EventStream, taskFailureRepository: TaskFailureRepo
 
     case AppTerminatedEvent(appId, eventType, timestamp) =>
```

Comment:
Upon receipt of an `AppTerminatedEvent`, the entry for the terminated app should be expunged also from the `taskOfferDeclinedRepository` here.